cltk/lemmatize/latin/unsupervised.py

```diff
@@ -0,0 +1,382 @@
+        the number of ambiguous cases in total; the number of tokens analyzed; and a
+        decimal between 0 and 1 representing the proportion of correct lemmatizations.
+        return
+        ------
+        a list object containing all incorrect lemmatizations for analysis. Format:
+        [(token, answer_given, correct_answer), (token...)]
+
+        NOTE: Initial tests show roughly 91% accuracy, identification of punctuation included.
+        '''
+        trials = 0
+        correct = 0
+        errors = []
+        for position in range(0, (len(token_list)-1)):
+            lemmalist = self.lemmatizer.lookup(token_list[position])
+            lemmalist = lemmalist[1]
+            lemma = max(lemmalist, key=itemgetter(1))
+            if len(lemmalist) > 1:
+                trials = trials + 1
+                if lemma[0] == lemma_list[position] or lemma[0] == 'punc':
+                    correct = correct + 1
+                else:
+                    errors.append((token_list[position], lemma[0], lemma_list[position]))
+        print(correct)
+        print(trials)
+        print(len(lemma_list))
+        rate = (len(lemma_list) - trials + correct) / len(lemma_list)
+        print(rate)
+        return errors
+
+
+class TessFile(object):
+    """Buffered/non-buffered reader for .tess file I/O.
+
+    Parameters
+    ----------
+    path : str
+        Path to the .tess file.
+    mode : str
+        File open mode ('r', 'w', 'a', etc.)
+    buffer : bool
+        If True, load file contents into memory on-the-fly. Otherwise, load in
+        contents on initialization.
+
+    Attributes
+    ----------
+    path : str
+        Path to the .tess file.
+    mode : str
+        File open mode ('r', 'w', 'a', etc.)
+    buffer : bool
+        If True, load file contents into memory on-the-fly. Otherwise, load in
+        contents on initialization.
+    hash : str
+        MD5 hash of the file.
+
+
+    """
+    def __init__(self, path, mode='r', buffer=True, validate=False):
+        self.path = path
+        self.mode = mode
+        self.buffer = buffer
+        self.fname = os.path.basename(path)
+
+        if buffer:
+            self.file = open(path, 'r')
+        else:
+            self.file = []
+            with open(path, 'r') as f:
+                for line in f.readlines():
+                    self.file.append(line)
+
+        self.__hash = None
+        self.__len = None
+
+        if validate:
+            self.validate()
+
+    def __getitem__(self, index):
+        if index < 0 or index >= len(self):
+            raise IndexError()
+
+        if self.buffer:
+            self.file.seek(0)
+            for _ in range(index + 1):
+                line = self.file.readline()
+            return line
+        return self.file[index]
+
+    def __len__(self):
+        if self.__len is None:
+            self.__len = sum([1 for _ in self.readlines()])
+        return self.__len
+
+    @property
+    def hash(self):
```

Comment:
This shadows the built-in Python hash function, at least in name collision. You should probably change it as a matter of habit and best practice.